sdk/keyvault/keyvault-keys/src/index.ts

```diff
@@ -479,34 +502,46 @@ export class KeysClient {
   /**
    * The delete operation applies to any key stored in Azure Key Vault. Individual versions
    * of a key can not be deleted, only all versions of a given key at once.
+   *
+   * This function returns a Long Running Operation poller that allows you to wait indifinetly until the key is deleted.
+   *
    * This operation requires the keys/delete permission.
    *
    * Example usage:
    * ```ts
-   * let client = new KeysClient(url, credentials);
-   * let result = await client.deleteKey("MyKey");
+   * const client = new KeysClient(url, credentials);
+   * const poller = await client.beginDeleteKey("MyKey");
+   *
+   * // Serializing the poller
+   * const serialized = poller.toJSON();
+   * // A new poller can be created with:
+   * // await client.beginDeleteKey("MyKey", { resumeFrom: serialized });
+   *
+   * // Waiting until it's done
+   * const deletedKey = await poller.pollUntilDone();
+   * console.log(deletedKey);
    * ```
    * @summary Deletes a key from a specified key vault.
-   * @param vaultBaseUrl The vault name, for example https://myvault.vault.azure.net.
    * @param name The name of the key.
    * @param [options] The optional parameters
    */
-  public async deleteKey(name: string, options?: RequestOptions): Promise<DeletedKey> {
-    const requestOptions = (options && options.requestOptions) || {};
-    const span = this.createSpan("deleteKey", requestOptions);
+  public beginDeleteKey(name: string, options?: BeginDeleteKeyOptions): DeleteKeyPoller {
+    // Making a fake client to reduce the duplicated code
+    const pollerClient: KeyClientInterface = {
```

Comment:
can't you just pass `this` as an instance of `KeyClientInterface` without doing this indirection? The real client should still satisfy the interface

Reply:
I'll try!

Reply:
We want to keep the original methods private :/

Reply:
Yeah, we talked about this offline. Looks like there's no good way to remove the private modifier in TS today, though there are a few proposal as to how this could be done. Boo classes!